keel-core/src/main/kotlin/com/netflix/spinnaker/keel/sync/Lock.kt

```diff
@@ -0,0 +1,7 @@
+package com.netflix.spinnaker.keel.sync
+
+import java.time.Duration
+
+interface Lock {
+  fun tryAcquire(name: String, duration: Duration): Boolean
```

Comment:
Do we ever want to release the lock early?

Reply:
Right now it's locking for one minute. I mean, yeah in theory, but I figure we can burn that bridge when we come to it.